src/components/Contacts.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import contacts from '../data/contacts.json'
+
+class Contacts extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: props.data.slice(0,5)
+    }
+  }
+
+  getRandom ()  {
+    console.log(this)
+    //make copy of contacts and splice randomContact so that it cannot be selected twice!!!
+    const contactsCopy = [...contacts]
+    const randomNum = Math.floor(Math.random() * contactsCopy.length);
+    const randomContact = contactsCopy.splice(randomNum, 1)
+    
+    this.setState({
+      data: [...this.state.data, randomContact[0]],
+    });
+  }
+
+  sortNames() {
+    this.setState({
```

Comment:
You could try to create on single function where you could pass the key that you would like to sort by and reused for `name` and `popularity`